app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java

```diff
@@ -175,30 +173,43 @@ public void onActivityResult(final int requestCode, final int resultCode,
 
         if ((requestCode == REQUEST_IMPORT_PATH || requestCode == REQUEST_EXPORT_PATH)
                 && resultCode == Activity.RESULT_OK && data.getData() != null) {
-            String path = Utils.getFileForUri(data.getData()).getAbsolutePath();
-            if (requestCode == REQUEST_EXPORT_PATH) {
-                SimpleDateFormat sdf = new SimpleDateFormat("yyyyMMdd_HHmmss", Locale.US);
-                exportDatabase(path + "/NewPipeData-" + sdf.format(new Date()) + ".zip");
-            } else {
-                AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
-                builder.setMessage(R.string.override_current_data)
-                        .setPositiveButton(getString(R.string.finish),
-                                (DialogInterface d, int id) -> importDatabase(path))
-                        .setNegativeButton(android.R.string.cancel,
-                                (DialogInterface d, int id) -> d.cancel());
-                builder.create().show();
+            try {
+                Uri uri = data.getData();
+                if (FilePickerActivityHelper.isOwnFileUri(getContext(), uri)) {
+                    uri = Uri.fromFile(Utils.getFileForUri(uri));
+                }
+                if (requestCode == REQUEST_EXPORT_PATH) {
+                    final StoredDirectoryHelper directory
+                            = new StoredDirectoryHelper(getContext(), uri, null);
+                    final SimpleDateFormat sdf
+                            = new SimpleDateFormat("yyyyMMdd_HHmmss", Locale.US);
+                    exportDatabase(directory.createFile("NewPipeData-"
+                            + sdf.format(new Date()) + ".zip", "application/zip"));
```

Comment:
Do we really want a directory picker here and not a file picker with a pre-filled name?

Reply:
See #3785

Reply:
A dir picker seems good to me

Reply:
@TobiGr: Why would you even prefer a directory picker over a file picker here?